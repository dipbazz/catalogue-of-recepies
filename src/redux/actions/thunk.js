import { fetchMeals, fetchMeal, fetchSearchMeals } from '../../API/api';
import {
  loadingMeal, errorMeal, successMeal, loadingMeals, errorMeals, successMeals,
} from './index';

const fetchMealsByCategory = (category) => (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchMeals(category);
  response
    .then((data) => dispatch(successMeals(data.meals)))
    .catch(() => dispatch(errorMeals('Error while fetching data.')));
};

const searchMeals = (query) => (dispatch) => {
  dispatch(loadingMeals());
  const response = fetchSearchMeals(query);
  response
    .then((data) => {
      if (data.meals) {
        dispatch(successMeals(data.meals));
      } else {
        dispatch(errorMeals(`There is no meal with name: ${query}`));
      }
    })
    .catch(() => dispatch(errorMeals('Error while fetching data.')));
};

const fetchMealById = (id) => (dispatch) => {
  dispatch(loadingMeal());
  const response = fetchMeal(id);
  response
    .then((data) => dispatch(successMeal(data.meals[0])))
    .catch(() => dispatch(errorMeal('Error while fetching data.')));
};

export { fetchMealsByCategory, searchMeals, fetchMealById };
