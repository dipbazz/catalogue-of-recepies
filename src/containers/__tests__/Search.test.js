import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import ConfigureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Search from '../Search';

const mockStore = ConfigureMockStore([thunk]);
jest.mock('../../API/api');

let store;
beforeEach(() => {
  store = mockStore({});
});

test('should test for search snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('should test for search submit event', () => {
  const handleSearch = jest.fn();
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    </Provider>,
  );

  const form = container.querySelector('form');
  const { search } = form.elements;
  search.value = 'Burger';
  // fireEvent.submit(search);
  fireEvent.keyDown(search, { key: 'enter', keyCode: 13 })

  expect(handleSearch).toHaveBeenCalledTimes(1);
});
