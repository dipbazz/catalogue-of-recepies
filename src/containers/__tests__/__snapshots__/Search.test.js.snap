// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test for search component snapshot 1`] = `
<form>
  <input
    class="search-input"
    name="search"
    placeholder="Search your recepies"
    type="search"
  />
</form>
`;
